### PR TITLE
:warning: metrics are now on by default

### PR DESCRIPTION
**Pull request by mengqiy**

### CODE REVIEW

File: pkg/metrics/listener.go [hunk 1]
```diff
@@ -21,11 +21,9 @@ import (
...
 )
 
-// DefaultBindAddress sets the default bind address for the metrics
-// listener
-// The metrics is off by default.
-// TODO: Flip the default by changing DefaultBindAddress back to ":8080" in the v0.2.0.
-var DefaultBindAddress = "0"
+// DefaultBindAddress sets the default bind address for the metrics listener
+// The metrics is on by default.
+var DefaultBindAddress = ":8080"
```
Comment (droot) [inline pkg/metrics/listener.go]: We should update the documentation for manager.Options https://godoc.org/sigs.k8s.io/controller-runtime/pkg/manager#Options explaining how to turn off or get the previous default behavior.

**Comment by droot** [comment]
A few more points that popped up during the conversation:
 - testenv pkg should disable metrics server by default so that tests can run in parallel by default
 - Make failure to "binding to default port 8080" obvious because people may encounter that in dev environment because 8080 being a popular port.

**Comment by mengqiy** [comment]
> testenv pkg should disable metrics server by default so that tests can run in parallel by default

testenv pkg handles the testing control plane, it doesn't touch `manager.Options`.
IMO it should be when we scaffold controller tests, metrics should be disabled in `manager.Options`.

**Comment by mengqiy** [comment]
PTAL

File: pkg/metrics/listener.go [hunk 2]
```diff
@@ -39,9 +37,12 @@ func NewListener(addr string) (net.Listener, error) {
 		return nil, nil
 	}
 
+	log.Info("metrics is starting to listen", "addr", addr)
```
Comment (droot) [inline pkg/metrics/listener.go]: `starting metrics server`

File: pkg/metrics/listener.go [hunk 3]
```diff
@@ -39,9 +37,12 @@ func NewListener(addr string) (net.Listener, error) {
...
 	ln, err := net.Listen("tcp", addr)
 	if err != nil {
-		return nil, fmt.Errorf("error listening on %s: %v", addr, err)
+		er := fmt.Errorf("error listening on %s: %v", addr, err)
+		log.Error(er, "metrics failed to listen. You may want to disable the metrics or use another port if it is due to conflicts")
```
Comment (droot) [inline pkg/metrics/listener.go]: May be reword it to say `metrics server failed to listen.` ?

File: pkg/manager/internal.go [hunk 4]
```diff
@@ -230,17 +230,19 @@ func (cm *controllerManager) GetWebhookServer() *webhook.Server {
...
 	// TODO(JoelSpeed): Use existing Kubernetes machinery for serving metrics
 	mux := http.NewServeMux()
-	mux.Handle("/metrics", handler)
+	mux.Handle(metricsPath, handler)
 	server := http.Server{
 		Handler: mux,
 	}
 	// Run the server
 	go func() {
+		log.Info("start to serve metrics", "path", metricsPath)
```
Comment (droot) [inline pkg/manager/internal.go]: May be reword it --> `starting metrics server` (And we can also add addr if available)  ?

**Comment by mengqiy** [comment]
PTAL

**Comment by vincepri** [comment]
We're getting testing failures because multiple processes are trying to bind port 8080.

PR: https://github.com/kubernetes-sigs/cluster-api/pull/1107
Test logs: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/sigs.k8s.io_cluster-api/1107/pull-cluster-api-test/1146217126005051403/

**Comment by mengqiy** [comment]
I added something in the GoDoc: https://github.com/kubernetes-sigs/controller-runtime/pull/510/files#diff-7d903b129bc883285ffc5068c39c00b0R139
I will update the release notes to make it more helpful.

**Comment by mengqiy** [comment]
Updated release notes and the description of this PR.

**Comment by vincepri** [comment]
Thank you!
